receiver/k8sclusterreceiver/collection/metadatastore.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collection
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// metadataStore keeps track of required caches exposed by informers.
+// This store is used while collecting metadata about Pods to be able
+// to correlate other Kubernetes objects with a Pod.
+type metadataStore struct {
+	stores map[string]cache.Store
+}
+
+// setupStore tracks metadata of services, jobs and replicasets.
+func (ms *metadataStore) setupStore(o runtime.Object, store cache.Store) {
+	switch o.(type) {
+	case *corev1.Service:
+		ms.stores["Service"] = store
+	case *batchv1.Job:
+		ms.stores["Job"] = store
+	case *appsv1.ReplicaSet:
```

Comment:
Does this need to be in a map? Seems like since all the possible stores are known at compile time (service, job, replicaset) they can just be entries in the struct?

Reply:
Also this struct feels a bit light in terms of what it's wrapping. Should the logic for getMetadataForX and the store be grouped together instead? I didn't dig into it too deep to feel strongly about it, just see what you think.

Reply:
I guess the stores for these informers could be fields on `collection.DataCollector` and those could be passed in to the methods that collect data? What do you think?

Reply:
Hmm oh yeah I forgot we were trying to keep code for a given resource together (ie getMetadtaForPods with pods). Unless you see something better I'd just keep how you have it but get rid of the map if runtime lookup isn't needed. ie:

`mc.stores["Job"]` would `become mc.jobs`

Reply:
Updated these stores to be fields on metadata store rather than a map.